webapp/src/components/news-list/NewsList.test.js

```diff
@@ -0,0 +1,162 @@
+import {describe, expect, it} from "@jest/globals";
+import {mount} from "@vue/test-utils";
+import NewsList from "@/components/news-list/NewsList";
+import NewsItem from "@/components/news-item/NewsItem";
+import NewsForm from "@/components/news-form/NewsForm";
+
+require('regenerator-runtime/runtime');
+
+  describe('NewsList', () => {
+    it('should show the message "The list is empty :(" when there are no elements in the list', () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [],
+              }
+          }
+      })
+      const paragraph = wrapper.find("h2");
+
+      expect(paragraph.text()).toEqual("The list is empty :(");
+      wrapper.destroy();
+  })
+
+  it("should show the elemens when they are present in the list", () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [
+                      {title: "Title_1", votes: 0},
+                      {title: "Title_2", votes: 0}
+                  ]
+              }
+          }
+      });
+
+      let children = wrapper.findAllComponents(NewsItem);
+      expect(children.length).toEqual(2);
+      wrapper.destroy();
+  })
+
+  it("should switch the sorting order whenever the button is clicked", () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [
+                      {title: "Title_1", votes: 5},
+                      {title: "Title_2", votes: 4}
+                  ],
+                  descending: true,
+              }
+          }
+      });
+
+      const button = wrapper.find('button.sorter');
+      button.trigger('click');
+      expect(wrapper.vm.descending).toEqual(false);
+      expect(wrapper.vm.sortedItems[0].votes).toEqual(4);
+
+      button.trigger('click');
+      expect(wrapper.vm.descending).toEqual(true);
+      expect(wrapper.vm.sortedItems[0].votes).toEqual(5);
+      wrapper.destroy();
+  })
+
+  it("should correctly update the votes of an item", async () => {
+      const wrapper = mount(NewsList, {
+          data() {
+              return {
+                  newsItems: [
+                      {title: "Title_1", votes: 5},
+                      {title: "Title_2", votes: 4}
+                  ],
+                  descending: true,
+              }
+          }
+      });
+
+      const item = wrapper.findComponent(NewsItem);
```

Comment:
You should test this feature in `NewsItem.test.js`, not in the parent component.